xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamBluray.cpp

```diff
@@ -147,6 +148,11 @@ bool CDVDInputStreamBluray::Open()
     root = url.GetHostName();
     filename = URIUtils::GetFileName(url.GetFileName());
 
+    // Check whether disc is AACS protected
+    CURL url3(root);
+    CFileItem base(url3, false);
+    openDisc = base.IsProtectedBlurayDisc();
+
```

Comment:
Sorry I must have missed this.

I don't know this code very well so my apologies if I'm asking dumb things here, but how would this be better since the m_item check below is already a FileItem, therefore IsProtectedBluray should be in CFileItem, no?

Also, I copied this logic from IsBluray/openDisc checks so why would we want this to be implemented differently than those two? Surely this isn't anymore overkill than those two, or am I missing something here? I mean IsBluray uses the exact same method for checking whether BDMV/index.bdmv exists (for example).

Reply:
I meant openStream checks. I mean IsBluray and IsDiscImage are both fileitems and I'm reusing the "item" and "m_item" which are fileitems already, so wouldn't moving IsProtectedBlurayDisc to URIUtils make this more inconsistent and complicated?